be/src/vec/exec/vjson_scanner.cpp

```diff
@@ -0,0 +1,729 @@
+#include "gutil/strings/split.h"
+#include "runtime/exec_env.h"
+#include "runtime/runtime_state.h"
+#include "util/time.h"
+
+namespace doris::vectorized {
+
+VJsonScanner::VJsonScanner(RuntimeState* state, RuntimeProfile* profile,
+                         const TBrokerScanRangeParams& params,
+                         const std::vector<TBrokerRangeDesc>& ranges,
+                         const std::vector<TNetworkAddress>& broker_addresses,
+                         const std::vector<TExpr>& pre_filter_texprs, ScannerCounter* counter)
+        : JsonScanner(state, profile, params, ranges, broker_addresses, pre_filter_texprs, counter),
+          _ranges(ranges),
+          _broker_addresses(broker_addresses),
+          _cur_file_reader(nullptr),
+          _cur_line_reader(nullptr),
+          _cur_vjson_reader(nullptr),
+          _next_range(0),
+          _cur_reader_eof(false),
+          _read_json_by_line(false) { 
+}
+
+VJsonScanner::~VJsonScanner() {
+    close();
+}
+
+Status VJsonScanner::open() {
+    RETURN_IF_ERROR(BaseScanner::open());
+    return Status::OK();
+}
+
+void VJsonScanner::close() { 
+    BaseScanner::close();
+    if (_cur_vjson_reader != nullptr) {
+        delete _cur_vjson_reader;
+        _cur_vjson_reader = nullptr;
+    }
+}
+
+Status VJsonScanner::get_next(vectorized::Block& output_block, bool* eof) {
+    SCOPED_TIMER(_read_timer);
+    Status status = Status::OK();
+    const int batch_size = _state->batch_size();
+    size_t slot_num = _src_slot_descs.size();
+    std::shared_ptr<vectorized::Block> temp_block(new vectorized::Block());
+    std::vector<vectorized::MutableColumnPtr> columns(slot_num);
+    auto string_type = make_nullable(std::make_shared<DataTypeString>());
+    for (int i = 0; i < slot_num; i++) {
+        columns[i] = string_type->create_column();
+    }
+
+    // Get one line
+    while (columns[0]->size() < batch_size && !_scanner_eof) {
+        if (_cur_file_reader == nullptr || _cur_reader_eof) {
+            RETURN_IF_ERROR(open_next_reader());
+            // If there isn't any more reader, break this
+            if (_scanner_eof) {
+                break;
+            }
+        }
+        
+        if (_read_json_by_line && _skip_next_line) {
+            size_t size = 0;
+            const uint8_t* line_ptr = nullptr;
+            RETURN_IF_ERROR(_cur_line_reader->read_line(&line_ptr, &size, &_cur_reader_eof));
+            _skip_next_line = false;
+            continue;
+        }
+
+        bool is_empty_row = false;
+        RETURN_IF_ERROR(_cur_vjson_reader->read_json_column(columns, _src_slot_descs,  
+                                                        &is_empty_row, &_cur_reader_eof));
+        if (is_empty_row) {
+            // Read empty row, just continue
+            continue;
+        }
+        COUNTER_UPDATE(_rows_read_counter, 1);
+        SCOPED_TIMER(_materialize_timer);
+    }
+
+    if (columns[0]->size() > 0) {
+        if (!_dest_vexpr_ctx.empty()) {
```

Comment:
This part of the logic for filtering and executing expressions should be abstracted to the base scanner so that other scanners can use it instead of writing it once